### PR TITLE
Add Differential ShellCheck starter workflow template

### PR DESCRIPTION
**Pull request by jamacku**

### CODE REVIEW

File: workflow-templates/differential-shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+
+name: Differential ShellCheck
+on:
+  push:
+  pull_request:
+    branches: [ $default-branch ]
```
Comment (phracek) [inline workflow-templates/differential-shellcheck.yml:7]: $default-branch is GitHub variable?

**Review by phracek — CHANGES_REQUESTED** [review]
I did not hit any issues. Thanks for the pull request. One comment.

File: workflow-templates/differential-shellcheck.yml (hunk 1 above)
Comment (jamacku) [reply workflow-templates/differential-shellcheck.yml:7]: Yes, but you can use it only in starter workflows (templates). From [GitHub docs](https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization#creating-a-starter-workflow):

_If you need to refer to a repository's default branch, you can use the `$default-branch` placeholder. When a workflow is created the placeholder will be automatically replaced with the name of the repository's default branch._

**Comment by jamacku** [comment]
I have also added a template for PyLint Action. @phracek PTAL.

**Review by phracek — APPROVED** [review]
THANKS. LGTM

**Pull request merged**